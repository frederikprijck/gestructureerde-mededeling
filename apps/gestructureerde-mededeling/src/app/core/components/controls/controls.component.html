<ng-container *ngIf="isSmallScreen$ | async; then smallControls; else largeControls"></ng-container>

<ng-template #smallControls>
  <div class="controls">
    <button
      *ngIf="refresh"
      mat-fab
      disableRipple
      color="primary"
      type="button"
      title="Verniewen"
      (click)="refreshClickHandler()"
    >
      <fa-icon [icon]="['fad', 'sync-alt']"></fa-icon>
    </button>
    <button
      *ngIf="copyNumber"
      class="ogm-copy"
      mat-fab
      disableRipple
      color="primary"
      type="button"
      title="Kopieer nummer"
      data-clipboard-action="copy"
      [attr.data-clipboard-text]="ogm?.number"
      (click)="copyNumberClickHandler()"
    >
      <fa-icon [icon]="['fad', 'comment-dots']"></fa-icon>
    </button>
    <button
      *ngIf="copyOgm"
      class="ogm-copy"
      mat-fab
      disableRipple
      color="primary"
      type="button"
      title="Kopieer mededeling"
      data-clipboard-action="copy"
      [attr.data-clipboard-text]="ogm?.numberFormat"
      (click)="copyOgmClickHandler()"
    >
      <fa-icon [icon]="['fad', 'comment-lines']"></fa-icon>
    </button>
  </div>
</ng-template>

<ng-template #largeControls>
  <div class="controls large">
    <button
      *ngIf="refresh"
      mat-raised-button
      isRoundButton
      disableRipple
      color="primary"
      (click)="refreshClickHandler()"
    >
      <fa-icon [icon]="['fad', 'sync-alt']"></fa-icon>Vernieuwen
    </button>
    <button
      *ngIf="copyNumber"
      class="ogm-copy"
      mat-raised-button
      isRoundButton
      disableRipple
      color="primary"
      data-clipboard-action="copy"
      [attr.data-clipboard-text]="ogm?.number"
      (click)="copyNumberClickHandler()"
    >
      <fa-icon [icon]="['fad', 'comment-dots']"></fa-icon>Kopieer nummer
    </button>
    <button
      *ngIf="copyOgm"
      class="ogm-copy"
      mat-raised-button
      isRoundButton
      disableRipple
      color="primary"
      data-clipboard-action="copy"
      [attr.data-clipboard-text]="ogm?.numberFormat"
      (click)="copyOgmClickHandler()"
    >
      <fa-icon [icon]="['fad', 'comment-lines']"></fa-icon>Kopieer mededeling
    </button>
  </div>
</ng-template>
