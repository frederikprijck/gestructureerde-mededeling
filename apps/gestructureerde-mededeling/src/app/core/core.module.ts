import { NgModule } from '@angular/core';
import { RoutingModule } from './routing.module';
import { MaterialModule } from './material.module';
import { ReactiveFormsModule } from '@angular/forms';

@NgModule({
  declarations: [],
  imports: [RoutingModule, MaterialModule, ReactiveFormsModule],
  exports: [RoutingModule, MaterialModule, ReactiveFormsModule],
})
export class CoreModule {}
