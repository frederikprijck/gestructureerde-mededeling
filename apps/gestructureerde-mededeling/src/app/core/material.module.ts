import { NgModule } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { LayoutModule } from '@angular/cdk/layout';
import { MatIconModule } from '@angular/material/icon';
import { MatCardModule } from '@angular/material/card';
import { MatButtonModule } from '@angular/material/button';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MatToolbarModule } from '@angular/material/toolbar';

@NgModule({
  declarations: [],
  imports: [
    BrowserAnimationsModule,
    MatButtonModule,
    MatCardModule,
    MatIconModule,
    LayoutModule,
    MatSnackBarModule,
    MatToolbarModule,
  ],
  exports: [MatButtonModule, MatCardModule, MatIconModule, LayoutModule, MatSnackBarModule, MatToolbarModule],
})
export class MaterialModule {}
