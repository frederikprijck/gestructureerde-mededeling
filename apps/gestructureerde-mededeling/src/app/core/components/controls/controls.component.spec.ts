import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ControlsComponent } from './controls.component';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { ScreenService } from '../../services/screen.service';

describe('ControlsComponent', () => {
  let component: ControlsComponent;
  let fixture: ComponentFixture<ControlsComponent>;
  let screenService: ScreenService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FontAwesomeModule],
      declarations: [ControlsComponent],
      providers: [ScreenService],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ControlsComponent);
    component = fixture.componentInstance;

    screenService = TestBed.inject(ScreenService);
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  describe('rendering component', () => {
    it('should match snapshot', () => {
      // act
      fixture.detectChanges();

      // check
      expect(fixture).toMatchSnapshot();
    });
  });

  describe('ngOnDestroy', () => {
    it('should call destroy method', () => {
      // prepare
      jest.spyOn(component.clipboard, 'destroy');

      // act
      component.ngOnDestroy();

      // check
      expect(component.clipboard.destroy).toHaveBeenCalled();
    });
  });

  describe('refreshClickHandler', () => {
    it('should emit refreshClick', () => {
      // prepare
      jest.spyOn(component.refreshClick, 'emit');

      // act
      component.refreshClickHandler();

      // check
      expect(component.refreshClick.emit).toHaveBeenCalled();
    });
  });

  describe('copyNumberClickHandler', () => {
    it('should emit refreshClick', () => {
      // prepare
      const message: string = 'Nummer gekopieerd naar klembord';
      jest.spyOn(component.copyNumberClick, 'emit');

      // act
      component.copyNumberClickHandler();

      // check
      expect(component.copyNumberClick.emit).toHaveBeenCalledWith(message);
    });
  });

  describe('copyOgmClickHandler', () => {
    it('should emit refreshClick', () => {
      // prepare
      const message: string = 'Mededeling gekopieerd naar klembord';
      jest.spyOn(component.copyOgmClick, 'emit');

      // act
      component.copyOgmClickHandler();

      // check
      expect(component.copyOgmClick.emit).toHaveBeenCalledWith(message);
    });
  });
});
