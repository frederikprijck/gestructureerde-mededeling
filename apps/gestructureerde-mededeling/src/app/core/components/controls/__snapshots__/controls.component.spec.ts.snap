// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsComponent rendering component should match snapshot 1`] = `
<ogm-controls
  clipboard={[Function Clipboard]}
  copyNumberClick={[Function EventEmitter]}
  copyOgmClick={[Function EventEmitter]}
  faIconLibrary={[Function FaIconLibrary]}
  isSmallScreen$={[Function Observable]}
  refreshClick={[Function EventEmitter]}
  screenService={[Function ScreenService]}
>
  <div
    class="controls large"
  >
    
    
    
  </div>
</ogm-controls>
`;
